php/libraries/NDB_BVL_Instrument_LINST.class.inc

```diff
@@ -599,6 +592,20 @@ class NDB_BVL_Instrument_LINST extends \NDB_BVL_Instrument
                                 );
                             }
                         } else {
+                            // addDateElement appends '_date' to the field name so
+                            // remove it first here
+                            if (strpos($pieces[1], "_date") !== false) {
+                                $pieces[1] = substr(
+                                    $pieces[1],
+                                    0,
+                                    strpos($pieces[1], "_date")
+                                );
+                            } else {
+                                throw new \LorisException(
+                                    "Standard Date format field $pieces[1] must end"
+                                      . " with '_date'."
+                                );
+                            }
```

Comment:
@zaliqarosli The problem with the placement of this code section is that it seems like the behaviour of the instrument builder is not consistent. I used the builder to create 8 date elements: 3 standards, 3 basics and 2 MY to test different usecases. here is the linst

```
table{@}bla
title{@}bla
date{@}Date_taken{@}Date of Administration{@}2006{@}2012
static{@}Candidate_Age{@}Candidate Age (Months)
static{@}Window_Difference{@}Window Difference (+/- Days)
select{@}Examiner{@}Examiner{@}NULL=>''
date{@}standard_date{@}STANDARD{@}1900{@}2000{@}
select{@}standard_date_status{@}{@}NULL=>''{-}'not_answered'=>'Not Answered'
date{@}basic{@}BASIC{@}1900{@}2000{@}BasicDate
date{@}my{@}MONTH YEAR{@}1900{@}2000{@}MonthYear
date{@}s_date{@}STANDARD DATE{@}1900{@}2000{@}Date
date{@}b_date{@}BASIC DATE{@}1900{@}2000{@}BasicDate
date{@}my_date{@}MY DATE{@}1900{@}2000{@}MonthYear
date{@}s_date_test{@}STANDARD DATE TEST{@}1900{@}2000{@}Date
date{@}b_date_test{@}BASIC DATE TEST{@}1900{@}2000{@}BasicDate
```

Notice how the first standard date has nothing as it's last element? but the other standard dates have `Date` as the last element ?

That means that in the case of the first standard date `standard_date` you will **NOT** go in the if statement on line 578. but in the case of the other 2 standard dates `s_date`  and `s_date_test` you **WILL** thus bypassing your current added logic.

If we don't want to fix/modify the instrument builder behaviour here, I would suggest adding `&& $dateFormat !== "Date"` to the if statement on line 578

Reply:
I think another improvment to bring here is to check that the _date is only at the end of the string, not anywhere in the name. I dont think the code should be stripping the first instance of `_date` in `I_added_a_date_element_date`, only the second instance

Reply:
hey @ridz1208 you are right for your first comment. in this PR https://github.com/aces/Loris/pull/6863#issuecomment-675752482 we would have to decide to either make the standard date format default "" or "Date". either way, for backward compatibility, i think it still makes sense to check for "Date" here